### PR TITLE
More py3 compat

### PR DESCRIPTION
**Pull request by leifj**
### All Submissions:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [ ] Have you added an explanation of what problem you are trying to solve with this PR?
* [ ] Have you added information on what your changes do and why you chose this as your solution?
* [ ] Have you written new tests for your changes?
* [x ] Does your submission pass tests?
* [ ] This project follows PEP8 style guide. Have you run your code against the 'flake8' linter?

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+idna==2.5
```
[the remark this answers is not in the document]
Comment (leifj) [reply requirements.txt]: I don't think we should pin at all. After my omnibus pyFF p3 PR is done there is no need for this and if somebody wants to build for p2 they can pin by explicitly installing those version before installing pyFF and pyXMLSecurity

File: src/xmlsec/crypto.py [hunk 2]
```diff
@@ -229,7 +232,7 @@ def sign(self, data, hash_alg=None):
             msg = r.json()
             if not 'signed' in msg:
                 raise ValueError("Missing signed data in response message")
-            return msg['signed'].decode('base64')
```
[the remark this answers is not in the document]
Comment (leifj) [reply src/xmlsec/crypto.py]: I changed to calling base64.encodestring - does that make sense in both cases?

File: requirements.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (johanlundberg) [reply requirements.txt]: Ok, but we agree that it should not be in requirements.txt then :)

File: src/xmlsec/test/p11_test.py [hunk 3]
```diff
@@ -244,7 +245,7 @@ def _get_all_signatures(t):
         sv = sig.findtext(".//{%s}SignatureValue" % xmlsec.NS['ds'])
         assert sv is not None
         # base64-dance to normalize newlines
-        res.append(sv.decode('base64').encode('base64'))
```
[the remark this answers is not in the document]
Comment (leifj) [reply src/xmlsec/test/p11_test.py]: huh? 

I rewrote to base64.encode/decodstring but... what do you mean "its not used" ?

[the remark this answers is not in the document]
Comment (leifj) [reply src/xmlsec/test/p11_test.py]: aha now I get it :-)

File: src/xmlsec/test/p11_test.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 import subprocess
 import shutil
 import tempfile
+import codecs
```
Comment (johanlundberg) [inline src/xmlsec/test/p11_test.py]: codecs is no longer used

**Pull request merged**